src/viam/app/app_client.py

```diff
@@ -0,0 +1,791 @@
+
+    async def get_robot_part_logs(
+        self,
+        robot_part_id: str,
+        filter: Optional[str] = None,
+        dest: Optional[str] = None,
+        errors_only: Optional[bool] = None,
+        num_pages: Optional[int] = None,
+    ) -> List[LogEntry]:
+        """Get the logs associated with a robot part.
+
+        Args:
+            robot_part_id (str): ID of the robot part to get logs from.
+            filter (Optional[str]): Only include logs with messages that contain the string `filter`. Defaults to empty string "" (i.e., no
+                filter).
+            dest (Optional[str]): Optional filepath to write the log entries to.
+            errors_only (Optional[bool]): Optional boolean specifying whether or not to only include error logs. Defaults to True.
+            num_pages: Optional number of pages of logs to return. Passing 0 returns all pages. Defaults to 1. All pages or the first
+                `num_pages` pages will be returned, whichever comes first.
+
+        Raises:
+            GRPCError: If an invalid robot part ID is passed.
+            AssertionError: If `num_pages` is less than 0.
+
+        Returns:
+            List[viam.app.app_client.LogEntry]: The list of log entries.
+        """
+        num_pages = num_pages if num_pages is not None else 1
+        assert num_pages >= 0
+        errors_only = errors_only if errors_only else True
+        page_token = ""
+        logs = []
+
+        if num_pages == 0:
+            while True:
+                next_page_token = await self._get_robot_part_logs(
+                    robot_part_id=robot_part_id, filter=filter, errors_only=errors_only, page_token=page_token, logs=logs
+                )
+                if not next_page_token or len(next_page_token) == 0:
+                    break
+                page_token = next_page_token
+        else:
+            while True:
+                next_page_token = await self._get_robot_part_logs(
+                    robot_part_id=robot_part_id, filter=filter, errors_only=errors_only, page_token=page_token, logs=logs
+                )
+                num_pages -= 1
+                if (not next_page_token or len(next_page_token) == 0) or num_pages == 0:
+                    break
+                page_token = next_page_token
+
+        if dest:
+            try:
+                file = open(dest, "w")
+                for log in logs:
+                    time = log.time
+                    level = log.level.upper()
+                    logger_name = log.logger_name.split(".")[0]
+                    file_name = log.caller["File"]
+                    line_number = int(log.caller["Line"])
+                    message = log.message
+                    file.write(f"{time}\t\t{level}\t{logger_name}\t{file_name}:{line_number}\t{message}\n")
+            except Exception as e:
+                LOGGER.error(f"Failed to write robot part logs to file {dest}", exc_info=e)
```

Comment:
Might be useful to have the robot part ID here as well, in case there are multiple log functions running